### PR TITLE
Upgrade to WGPU 22

### PR DESCRIPTION
**Pull request by callym**
This currently doesn't pass the tests, I think because `naga` removed the `PartialEq` implementation on a bunch of types (https://github.com/gfx-rs/wgpu/pull/5818)

### CODE REVIEW

**Comment by Elabajaba** [comment]
I haven't ran it locally, but the test output tends to change a bit on wgpu/naga upgrades.

**Comment by teoxoy** [comment]
I put up https://github.com/gfx-rs/wgpu/pull/6045 that will be backported and part of the 22.1 release.

**Comment by JMS55** [comment]
Naga 22.1 has released with PartialEq back on types.

**Comment by callym** [comment]
I've updated some failing tests - to me it looks like just the numbering in the error message is different

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ prune = []
 allow_deprecated = []
 
 [dependencies]
-naga = { version = "0.20", features = ["wgsl-in", "wgsl-out"] }
+naga = { version = "22", features = ["wgsl-in", "wgsl-out"] }
```
Comment (JMS55) [inline Cargo.toml:20]: We should depend on 22.1 explicitly.

**Review by JMS55 — CHANGES_REQUESTED** [review]

**Comment by callym** [comment]
The test failures are assertions that were never hit before because they used `#[cfg(debug)]` instead of `#[cfg(debug_assertions)]`

**Review by JMS55 — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -31,6 +31,6 @@ once_cell = "1.17.0"
 indexmap = "2"
 
 [dev-dependencies]
-wgpu = { version = "0.20", features = ["naga-ir"] }
+wgpu = { version = "22", features = ["naga-ir"] }
```
Comment (superdump) [inline Cargo.toml:34]: Also 22.1?

**Review by superdump — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request merged**